src/twisted/web/_stan.py

```diff
@@ -23,13 +23,13 @@
 
 
 from inspect import isgenerator, iscoroutine
-from typing import TYPE_CHECKING, Dict, List, Optional, Union
+from typing import Dict, List, Optional, TYPE_CHECKING, Union
 from warnings import warn
 
 import attr
 
 if TYPE_CHECKING:
-    from twisted.web.template import Flattenable
+    from ._flatten import Flattenable
```

Comment:
Why this change? `Flattenable` is explicitly exported from `twisted.web.template` via `__all__`. Importing it from its public location should be more future proof than importing it from its implementation module.

Reply:
That usually makes cycles worse

Reply:
I think using `from . import _flatten` and then `x: "_flatten.Flattenable"` is preferable to `if TYPE_CHECKING`

Reply:
`mypy` doesn't care about import cycles, so that's only a factor if you want to remove the `if TYPE_CHECKING`. Even if you import from the implementation module, you'll still have to keep import cycles in mind, so I don't think removing `if TYPE_CHECKING` will simplify the code.